influxdb3_write/src/write_buffer/table_buffer.rs

```diff
@@ -291,6 +446,44 @@ impl BufferIndex {
         Self { columns }
     }
 
+    fn new_from_data(data: &BTreeMap<String, Builder>, old_index: &BufferIndex) -> Self {
+        let columns = data
+            .iter()
+            .filter_map(|(c, b)| {
+                if old_index.columns.contains_key(c) {
+                    let mut column: HashMap<String, Vec<usize>> = HashMap::new();
+                    match b {
+                        Builder::Tag(b) => {
```

Comment:
`Builder::Key` can be added to this arm. They use the same builder so it should just be
```rust
Builder::Tag(b) | Builder::Key(b) => {
    /* ... */
```